src/header/map.rs

```diff
@@ -1631,6 +1543,129 @@ impl<T> HeaderMap<T> {
         let more = self.capacity() - self.entries.len();
         self.entries.reserve_exact(more);
     }
+
+    #[inline]
+    fn raw_links(&mut self) -> RawLinks<T> {
+        RawLinks(&mut self.entries[..] as *mut _)
+    }
+}
+
+/// Removes the `ExtraValue` at the given index.
+#[inline]
+fn remove_extra_value<T>(mut raw_links: RawLinks<T>, extra_values: &mut Vec<ExtraValue<T>>, idx: usize) -> ExtraValue<T> {
```

Comment:
Nit/TIOLI: seems like this could be a method on `RawLinks`? Not a big deal regardless.